### PR TITLE
doc(otel): add main.dox

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #11156 

- Add a main page for `google/cloud/opentelemetry`
- Link to it from the README
- Add a brief description to the README

I want to get this in place before we add samples. (so we can reference those samples from the index page)

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11692)

### CODE REVIEW

File: google/cloud/opentelemetry/README.md [hunk 1]
```diff
@@ -90,10 +94,12 @@ int main(int argc, char* argv[]) {
 
 - If you want to learn more about the [OpenTelemetry] for C++ SDKs, consult
```
Comment (coryan) [inline google/cloud/opentelemetry/README.md:95]: In other libraries we removed the `Build and Install` section in favor of a link to https://cloud.google.com/cpp/docs/setup/

File: google/cloud/opentelemetry/README.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
-# C++ OpenTelemetry Exporters for Google Cloud Platform services.
+# Google Cloud C++ OpenTelemetry Exporters
+
+This directory contains the [OpenTelemetry] [trace exporter][trace-exporter] for
+[Cloud Trace][cloud-trace]. It also contains helpers to install the Cloud Trace
```
Comment (coryan) [inline google/cloud/opentelemetry/README.md:4]: nit: I think this paragraph assumes the reader knows what a trace exporter is, and why one may want one of them.  Maybe say:

```
This directory contains components to export [traces] captured with OpenTelemetry to [Cloud Trace].
Traces are a form of instrumentation where calls across multiple distributed systems can be collected
in a centralized view, showing the sequence of calls and their latency. [Cloud Trace] can provide such
views of your system calls.

OpenTelemetry offers APIs to capture traces. The C++ client libraries are already instrumented, you can use this instrumentation with minimal changes to your application. It is not necessary to instrument your application too.
```

Feel free to massively edit or expand that.  Just assume the reader is learning about tracing for the first time.

File: google/cloud/opentelemetry/doc/main.dox [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*!
+
+@mainpage Google Cloud C++ OpenTelemetry Exporters
+
+This directory contains the [OpenTelemetry] [trace exporter][trace-exporter] for
```
Comment (coryan) [inline google/cloud/opentelemetry/doc/main.dox:5]: same comments as the README.

**Review by coryan — APPROVED** [review]
A few nits and suggestions, but merging AS-IS is good too, better than what we currently have.

**Review by coryan — APPROVED** [review]
A few nits and suggestions, but merging AS-IS is good too, better than what we currently have.

**Review by coryan — APPROVED** [review]
A few nits and suggestions, but merging AS-IS is good too, better than what we currently have.

**Review by coryan — APPROVED** [review]
A few nits and suggestions, but merging AS-IS is good too, better than what we currently have.

File: google/cloud/opentelemetry/README.md [hunk 4]
```diff
@@ -1,4 +1,8 @@
-# C++ OpenTelemetry Exporters for Google Cloud Platform services.
+# Google Cloud C++ OpenTelemetry Exporters
+
+This directory contains the [OpenTelemetry] [trace exporter][trace-exporter] for
```
[the remark this answers is not in the document]
Comment (dbolduc) [reply google/cloud/opentelemetry/README.md]: Done.

File: google/cloud/opentelemetry/doc/main.dox (hunk 3 above)
Comment (dbolduc) [reply google/cloud/opentelemetry/doc/main.dox]: Ack.

**Pull request merged**